### PR TITLE
feat(ifo 3.1): Add Private Sale

### PR DESCRIPTION
**Pull request by chefilip**

### CODE REVIEW

File: src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IfoCardTokens.tsx [hunk 1]
```diff
@@ -182,6 +187,23 @@ const IfoCardTokens: React.FC<IfoCardTokensProps> = ({
       )
     }
 
+    const ifov31Msg =
+      ifo.version === 3.1 && poolId === PoolIds.poolBasic && criterias?.length > 0 ? (
+        <Box mt="16px">
+          {!isEligible && (
+            <Message mx="24px" mb="24px" p="8px" variant="danger">
+              <MessageText display="inline">Meet any one of the following requirements to be eligible.</MessageText>
```
Comment (memoyil) [inline src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IfoCardTokens.tsx:195]: Missing translation

File: src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IfoCardTokens.tsx [hunk 2]
```diff
@@ -182,6 +187,23 @@ const IfoCardTokens: React.FC<IfoCardTokensProps> = ({
...
+            </Message>
+          )}
+          <IFORequirements criterias={criterias} pointThreshold={publicPoolCharacteristics?.pointThreshold} />
+          {isEligible && (
+            <Message mx="24px" mt="24px" p="8px" variant="success">
+              <MessageText display="inline">You are eligible to participate in this Private Sale!</MessageText>
```
Comment (memoyil) [inline src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IfoCardTokens.tsx:201]: Missing translation

File: src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/index.tsx [hunk 3]
```diff
@@ -31,13 +33,44 @@ interface IfoCardProps {
...
+        const msgs = Object.keys(meta)
+          .filter((criteria) => meta[criteria])
+          .map((criteria) => MSG_MAP[criteria])
+
+        return {
+          title: t('Private Sale'),
+          variant: 'blue',
+          tooltip: msgs?.length ? (
+            <>
+              <Text>Meet any one of the requirements to join:</Text>
```
Comment (memoyil) [inline src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/index.tsx:63]: Translation

File: src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IfoCardTokens.tsx (hunk 1 above)
Comment (chefilip) [reply src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IfoCardTokens.tsx:195]: Yeah. Thanks. It's just a draft. I will add translation later.

File: src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IFORequirements.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  Box,
+  useTooltip,
+  LinkExternal,
+} from '@pancakeswap/uikit'
+import { useWeb3React } from '@web3-react/core'
+import { formatBigNumber } from 'utils/formatBalance'
+
+const NotOkNFT = () => (
+  <>
+    Set{' '}
```
Comment (memoyil) [inline src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IFORequirements.tsx:18]: Can we first translate the full text then split it with 'Pancake Squad NFT' (that won't be translated) then use split text array to construct. In that way we will have more translation friendly text

File: src/views/Ifos/hooks/v3/useCriteras.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react'
+
+const mapCriteriasToQualifications = {
+  needQualifiedNFT: 'isQualifiedNFT',
+  needQualifiedPoints: 'isQualifiedPoints',
+}
+
+export default function useCriteras(userBasicPoolInfo, ifoCriterias) {
```
Comment (memoyil) [inline src/views/Ifos/hooks/v3/useCriteras.ts:8]: Typo

File: src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IFORequirements.tsx (hunk 4 above)
Comment (chefilip) [reply src/views/Ifos/components/IfoFoldableCard/IfoPoolCard/IFORequirements.tsx:17]: It's a good idea. I will implement it when I add translation

**Review by chef-jojo — APPROVED** [review]
lgtm

**Review by Chef-Cheems — APPROVED** [review]

**Pull request merged**